GLOSSARY.md

```diff
@@ -0,0 +1,284 @@
+--------------- | ------------------ | ------------------
+**Expected**    | :no_entry_sign:    | :white_check_mark:
+**Unexpected**  | :white_check_mark: | :white_check_mark:
+
+## Post Mortem Analysis
+
+The practice of debugging a program after it has already crashed by inspecting
+the serialized state of the program at the time of error. This usually leverages
+tools like `lldb`, `gdb`, or `mdb` to inspect a "core" (defined below.)
+
+### Abort
+
+The [`abort()`][abort-syscall] system call. Accessible to Node via
+`process.abort()`. Called by V8 when `--abort-on-uncaught-exception` is enabled
+and an exception is thrown without a top-level `catch` handler.
+
+Halts the program and serializes the state of the program (the contents of the
+stack and heap segments, as well as the state of open file descriptors) to a
+"core file." Causes the program to return a non-zero exit status.
+
+### Core / Core dump / Core file
+
+The serialized state of a program, containing the stack and heap segments of
+memory in their entirety. Analyzable via `mdb`, `gdb`, or `lldb`. V8-specific
+plugins are available in the form of `mdb_v8`, `llnode`, and `lldb-v8`, and
+allow users to inspect JavaScript stack frames as well as the frozen object
+graph in memory (including objects pending garbage collection.)
+
+### Stack Frame
+
+A data structure held in the stack segment of memory representing a function
+call. This typically saves the arguments with which the function was called
+as well as a return address to jump to when the call completes.
+
+### Unwinding the Stack / Top of Stack
+
+Stack frames are collected into a stack that grows and shrinks as functions
+are called and functions return. Exceptions work by "unwinding" the stack
+to the nearest exception handler, if any — that is, they remove stack frames
+giving control back to the nearest frame with an exception handler. The
+stack is grown and shrunk in-place, so any subsequent function calls (even
+ones implicitly made by the JS VM) will clobber the original stack leading
+to the error. `Error` stacktraces _record_ some frame information leading to
+the error, but throwing an error is generally a lossy operation.
+
+If no exception handler is registered in a stack, instead of unwinding the
+stack, V8 may abort the program or call a user-defined `OnUncaughtException`
+handler.
+
+## Promise
+
+The 30-second version:
+
+A [well-specified][promises-aplus] pattern that provides a container type
+designed to describe the dependence graph of asynchronous operations within a
+program. As a container type, any value `T` may be cast to an asynchronous
+value `Promise<T>`. The process of casting `T` to `Promise<T>` is known as
+"resolution". Any value `Promise<T>` may be unwrapped to `T` by passing a
+handler (defined below) to `.then`. Unwrapping via `.then` creates a new
+`Promise` that will resolve to the return value of the handler. One promise
+may have zero-to-many dependent promises.
```

Comment:
I'm not sure parent-child is a good termonology and it hasn't emerged in the community over the last couple of years.

I do agree we need termionology for this. Maybe @erights who formalized these concepts have a better terminology?

Also @kriskowal it would be great if you could do a terminology sweep here.